### PR TITLE
add assignment_1

### PR DESCRIPTION
**Pull request by t-kanin**

### CODE REVIEW

File: assignment_1.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+def solution(number) 
+
+	arr = Array.new
```
Comment (dolwijit13) [inline assignment_1.rb]: Not sure if tab size is 2 or not?

File: assignment_1.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+def solution(number) 
+
```
Comment (dolwijit13) [inline assignment_1.rb]: Remove empty line

File: assignment_1.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+p test1 = solution(9687) 
+p test2 = solution(2000)
+p test3 = solution(5445)
+
+# test 1 digit 
+p test4 = solution(3)
+p test0 = solution(0) 
+
+# test 2 digits
+p test5 = solution(45)
```
Comment (WatcharitNew) [inline assignment_1.rb]: May add the right answer for each case to confirm correct answer

File: assignment_1.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  return arr << number if number < 10
+
+  while number != 0 do 
+    arr << number % 10 
+    number = number / 10 
+  end 
+
+  arr = arr.sort.reverse
+  res = arr.join.to_i
+  res
```
Comment (jittarinkan) [inline assignment_1.rb]: Maybe just arr.join.to_i. Don't need to assign it to res.

**Comment by aunsira** [comment]
Incorrect spaces.

File: assignment_1.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+p test0 = solution(0) #expect 0 got 0 
+
+# test 2 digits
+p test5 = solution(45) # expect 54 got 54
+
+
+
+
+
+
```
Comment (aunsira) [inline assignment_1.rb]: Please cleanup these  extra lines.

File: assignment_1.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+def solution(number) 
+  arr = []
+  return number if number < 10
+
+  while number != 0 do
```
Comment (aunsira) [inline assignment_1.rb]: Ruby provides `zero?` to check whether given number is zero or not. https://ruby-doc.org/core-2.7.3/Numeric.html#zero-3F-method

File: assignment_1.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  arr = []
+  return number if number < 10
+
+  while number != 0 do 
+    arr << number%10 
+    number
+  end 
+	
+  arr_sort = arr.sort.reverse
+  arr_sort.join.to_i
```
Comment (aunsira) [inline assignment_1.rb]: Actually we don't need `arr_sort` if we join these 2 lines together.

**Review by aunsira — APPROVED** [review]

**Pull request merged**